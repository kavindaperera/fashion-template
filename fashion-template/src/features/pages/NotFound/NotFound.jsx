import React, { Component } from "react";
import "react-responsive-carousel/lib/styles/carousel.min.css";
import _ from "lodash";
import "./NotFound.css";
import { Helmet } from "react-helmet";
class NotFound extends Component {
  render() {
    //console.log('404')
    return (
      <body  data-testid="not-found" id="error">
            <Helmet>
              <title>Error 404</title>
            </Helmet>
        <div id="notfound">
          <div className="notfound">
            <div className="notfound-404">
              <h1>404</h1>
              <h2>Page not found</h2>
            </div>
            <a href="javascript:history.back()">Go Back</a>
          </div>
        </div>
      </body>
    );
  }
}

export default NotFound;
