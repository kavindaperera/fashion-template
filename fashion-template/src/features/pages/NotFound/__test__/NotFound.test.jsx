import React from 'react';
import ReactDOM from 'react-dom';
import NotFound from "../NotFound";
import { render, cleanup } from '@testing-library/react'
import "@testing-library/jest-dom/extend-expect"
import renderer from 'react-test-renderer'

afterEach(cleanup);

it("renders without crashing", ()=>{
    const div = document.createElement("div");
    ReactDOM.render(<NotFound></NotFound>,div)
});

it("renders page correctly",()=>{
    const {getByTestId} = render(<NotFound ></NotFound>)
    expect(getByTestId('not-found')).toHaveTextContent('404Page not foundGo Back')
});

it('matches snapshot 1', ()=>{
    const tree = renderer.create(<NotFound></NotFound>).toJSON();
    expect(tree).toMatchSnapshot();
})