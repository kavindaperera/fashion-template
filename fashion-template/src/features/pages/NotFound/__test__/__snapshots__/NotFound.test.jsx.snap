// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<body
  data-testid="not-found"
  id="error"
>
  <div
    id="notfound"
  >
    <div
      className="notfound"
    >
      <div
        className="notfound-404"
      >
        <h1>
          404
        </h1>
        <h2>
          Page not found
        </h2>
      </div>
      <a
        href="javascript:history.back()"
      >
        Go Back
      </a>
    </div>
  </div>
</body>
`;
